### PR TITLE
changing links to our repo

### PR DESCRIPTION
**Pull request by darkdarkdragon**
In preparation to be used in LPMS

### CODE REVIEW

**Comment by darkdarkdragon** [comment]
@j0sh Accidentally I've added here `livepeer` branch. How do you think we should maintain code here? Should we work off the `master` branch and use it in `LPMS` or use separate `livepeer` branch and use it in LMPS?

File: av/avconv/avconv.go
Comment (mkrufky) [inline av/avconv/avconv.go]: whitespace breakage - was this added space intentional?

**Review by mkrufky — COMMENTED** [review]
In the title of this PR, should `chasing` be `changing?`  If so, please fix that :-)  :heavy_check_mark: 

I see three separate changes in this PR:
* b9ea2c9 - ￼use a 10mb chunk size
* 7139c5a - ￼change all the links to point to Livepeer's repo
* 7139c5a - ￼linter / whitespace cleanups

First off, could you explain the first change?  I see the value in a smaller "chunk size" but I think the reason for the change should be reflected in the commit log.

Rather than hardcoding this value, might it make sense to make it configurable?

Next, what are we actually doing here?  Are we "special sauce"ing the smaller chunk size?  Is livepeer taking over maintenance of the joy4 project?  I am worried that this change is going to make things difficult as far as staying in line with upstream.  Is this fork really necessary?

Please think about upstreaming the whitespace change separately, and perhaps making the chunksize a configurable option.

For the sake of future cherry-picking, in case we ever rejoin mainline joy4, please rebase these as individual commits (exactly as-is) when merging.

**Comment by j0sh** [comment]
Is this wholesale renaming from nareix to livepeer really necessary given that we're vendoring joy4 with the goclient? (Not sure how this would affect go modules.)

...On second glance, I see how this things weird with LPMS. That seems rather unpleasant.

>  How do you think we should maintain code here? Should we work off the master branch and use it in LPMS or use separate livepeer branch and use it in LMPS?

I'd like to remain explicit about the fact that we're working off a Livepeer-specific branch of joy4, so having a `livepeer` branch seems OK, but open to other options especially if we don't have a choice but to rename all the internal imports.

**Comment by mkrufky** [comment]
@j0sh , Explicit hash / branch dependency is supported in `go mod` ...  If we can remove the `gx` / `ipfs` dependencies from go-livepeer, integrating a branch dependency for a small fix like this becomes very easy.

**Comment by darkdarkdragon** [comment]
@mkrufky 
> In the title of this PR, should chasing be changing? If so, please fix that :-)

Was in hurry, fixed.

> use a 10mb chunk size

This was @iameli's commit, don't know anything about it. As I've pointed out in upper comment, I've just missed the fact that it was in different branch.

> I'd like to remain explicit about the fact that we're working off a Livepeer-specific branch of joy4, so having a livepeer branch seems OK

I think that it is good idea too, just wanted to be sure we how to handle things here.

@mkrufky @j0sh I've missed fact that there was whitespace changes - VS Code automatically reformatted it. I've reversed whitespace changes, so there is only links changes now in commit.

**Comment by mkrufky** [comment]
I confirm that now this PR only changes all instances of `nareix` to `livepeer`, but the question remains: Why are are we doing this?  Is livepeer taking over maintenance of the joy4 project?  Is this fork really necessary?

**Comment by j0sh** [comment]
> Is livepeer taking over maintenance of the joy4 project? Is this fork really necessary?

The situation is something like this:

* joy4 is unmaintained
* We maintain a separate `livepeer` branch in the `livepeer/joy4` repo with Livepeer-specific fixes
* We vendor joy4 into go-livepeer, which would make this rename unnecessary for the goclient itself
* LPMS does not vendor joy4, which makes things a bit problematic: folks using LPMS standalone would have to manually update their joy4 version to the latest `livepeer` branch in the livepeer/joy4 repo. This includes Livepeer team members who are developing on LPMS.

There's a few options I can think of for this situation: 
1. This renaming
2. Not renaming but maintaining a livepeer branch only with specific fixes and expecting folks to manually pull / update joy4 as necessary
3. Converting LPMS to go-modules as a first step towards goclient go-module conversion (how do go modules compose across different projects?)

Personally, I think I'm in favor of minimizing churn, so option 1 doesn't seem desirable to me. Option 3 (using goclient modules in LPMS) might be a good opportunity to start updating the codebase with current golang best practices. If that's too complicated, option 2 seems OK in the interim until we have the bandwidth for option 3.

Maybe there's another alternative?

> If we can remove the gx / ipfs dependencies from go-livepeer

That's a separate issue, but currently IPFS is disabled and I don't see us re-enabling IPFS anytime soon, so I'd be OK with removing it. Probably should make another ticket in the go-livepeer repo to solicit feedback from others first though.

**Review by mk-livepeer — APPROVED** [review]
I also noticed that @ericxtang has merge rights on the original `nareix/joy4` repo...

Perhaps Livepeer *is* taking over maintenance of this repository, or at least this one fork.

I have already converted `joy4` to use go-mod locally -- I'll push up a PR after this is merged.

I think we should start moving to `go mod` *and* merge this now, as this will help in that effort.

If upstream `nareix/joy4` becomes active again, we'll just have to rebase this tree and remove this particular change at that point.

This does seem like the right thing to do for now, after all.

**Comment by j0sh** [comment]
If we convert LPMS to go mod, do we still need this change?

Both Eric and I have commit rights to joy4 but we don't want to use it just
to push Livepeer changes through and leave community patches on the
sidelines. Ivan's patch fixing our issue should go in as a PR though, just
so it's visible.

**Comment by mkrufky** [comment]
This PR lands in the `livepeer` branch.  It seems right to merge it there.

Otherwise, the `go.mod` dependency tree in the current state looks like this:
```go.mod
module github.com/livepeer/joy4

go 1.12

require github.com/nareix/joy4 v0.0.0-20181022032202-3ddbc8f9d431
```
but after merging Ivan's PR the `go.mod` dependency tree looks like this:
```go.mod
module github.com/livepeer/joy4

go 1.12
```

**Comment by mkrufky** [comment]
This PR lands in the `livepeer` branch.  It seems right to merge it there.

Otherwise, the `go.mod` dependency tree in the current state looks like this:
```go.mod
module github.com/livepeer/joy4

go 1.12

require github.com/nareix/joy4 v0.0.0-20181022032202-3ddbc8f9d431
```
but after merging Ivan's PR the `go.mod` dependency tree looks like this:
```go.mod
module github.com/livepeer/joy4

go 1.12
```

Without changing the imports, we're not actually using this code - we need to change the imports otherwise we're still building the upstream code rather than our fork.

**Comment by darkdarkdragon** [comment]
> Without changing the imports, we're not actually using this code - we need to change the imports otherwise we're still building the upstream code rather than our fork.

I've played a bit with `go mod` - it actually allows us to use forked repo without changing links internally (`replace` command in `go.mod`) - but it works only on root repo (root repo is `go-livepeer` in our case, but we want `lpms` to use forked `joy4` - so it won't work in our case.

> joy4 is unmaintained

When we should just maintain our own fork with own branch.

**Comment by j0sh** [comment]
> root repo is go-livepeer in our case, but we want lpms to use forked joy4

Couldn't we have a go.mod for LPMS itself in addition to the goclient's go.mod?

I know nothing about `go mod` but this restriction seems really unfortunate, as that makes it hard to maintain out-of-tree changes with minimal disruption to the underlying code. There may be occasions a few patches on top of some other project is needed, while still tracking the project on an ongoing basis (eg, a workflow similar to Debian's `quilt`).

> When we should just maintain our own fork with own branch.

That's what we've been doing, but there is still value in soft forks rather than "hard" forks (as this is). I had thought about us keeping a permanent PR open to joy4 pointing to the `livepeer` branch for visibility and as a courtesy to the community, but that becomes unappealing with this rename.

**Comment by darkdarkdragon** [comment]
>  restriction seems really unfortunate, as that makes it hard to maintain out-of-tree changes with minimal disruption to the underlying code.

It's problem of `golang` design.

> There may be occasions a few patches on top of some other project is needed,

`go mod` adds hack to solve just this problem, but our use case slightly different.
`go mod` allows to specify in `go-livepeer` that our fork of `joy4` should be used, but doesn't allow to put such instruction into `lpms` (`lmps` can't specify that fork of `joy4` should be used for building `go-livepeer`.

>  permanent PR open to joy4 pointing to the livepeer branch for visibility and as a courtesy to the community

Don't know about others, but I'm personally, when interested in project and I see it doesn't have recent commit - I'm looking into Github's `network` graph to find out recent additions/fixes.

**Comment by j0sh** [comment]
> lmps can't specify that fork of joy4 should be used for building go-livepeer.

If `go mod` doesn't do transitive dependency resolution, that seems OK for us. Can we can have 
one `go.mod` for go-livepeer, and another `go.mod` for LPMS for when we're developing LPMS standalone?

> I'm looking into Github's network graph to find out recent additions/fixes

Even then, this type of change may make it harder for others for to cherry-pick fixes or quickly switch to our branch in their code for testing. It used to be a matter of git-fetching and changing branches; now it'll also involve actual changes to the imports.

**Comment by darkdarkdragon** [comment]
> Can we can have
one go.mod for go-livepeer, and another go.mod for LPMS for when we're developing LPMS standalone?

We can. But everyone else who uses `lpms` as a lib will miss it.

> Even then, this type of change may make it harder for others for to cherry-pick fixes

Cherry picking shouldn't be problem because we have one separate commit for import renames.

>  quickly switch to our branch in their code for testing.

`go mod` solves this problem. But this solution is just for quick testing/temporary hack, making use of it for permanent development of such a big project will add a lot of confusion.

**Pull request merged**
